src/query/api/v1/handler/graphite/find.go

```diff
@@ -57,52 +59,94 @@ func NewFindHandler(
 	}
 }
 
+func mergeTags(
+	terminatedResult *storage.CompleteTagsResult,
+	childResult *storage.CompleteTagsResult,
+) (map[string]bool, error) {
+	// sanity check the case.
+	if terminatedResult.CompleteNameOnly {
+		return nil, errors.New("terminated result is completing name only")
+	}
+
+	if childResult.CompleteNameOnly {
+		return nil, errors.New("child result is completing name only")
+	}
+
+	mapLength := len(terminatedResult.CompletedTags) + len(childResult.CompletedTags)
+	tagMap := make(map[string]bool, mapLength)
```

Comment:
nit: In future we should use an autogen'd map that can use `[]byte` as the key and we could then use `SetUnsafe(key, ..., Options{NoCopyKey: true, NoFinalizeKey: true})`.